### PR TITLE
[WIP] Checking Master with  <XunitShowProgress>true</XunitShowProgress> Outerloop

### PR DESCRIPTION
**Pull request by Drawaes**

### CODE REVIEW

**Comment by Drawaes** [comment]
@dotnet-bot Test Outerloop Linux x64 Release Build

File: src/System.Net.Security/tests/FunctionalTests/System.Net.Security.Tests.csproj [hunk 1]
```diff
@@ -3,6 +3,7 @@
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
   <PropertyGroup>
     <ProjectGuid>{A55A2B9A-830F-4330-A0E7-02A9FB30ABD2}</ProjectGuid>
+    <XunitShowProgress>true</XunitShowProgress>
```
Comment (danmosemsft) [inline src/System.Net.Security/tests/FunctionalTests/System.Net.Security.Tests.csproj]: I think you should add <XunitMaxThreads>1</XunitMaxThreads> also to make the output not be interleaved.

**Comment by stephentoub** [comment]
```
2017-10-21 04:45:44,685: INFO: proc(54): run_and_log_output: Output:    System.Net.Security.Tests.SslStreamStreamToStreamTest_BeginEnd.SslStream_StreamToStream_Authentication_IncorrectServerName_Fail [STARTING]
2017-10-21 04:45:44,797: INFO: proc(54): run_and_log_output: Output: 
2017-10-21 04:45:44,798: INFO: proc(54): run_and_log_output: Output: Unhandled Exception: System.NullReferenceException: Object reference not set to an instance of an object.
2017-10-21 04:45:44,798: INFO: proc(54): run_and_log_output: Output:    at System.Net.LazyAsyncResult.get_CurrentThreadContext() in /mnt/j/workspace/dotnet_corefx/master/linux-TGroup_netcoreapp+CGroup_Release+AGroup_x64+TestOuter_false_prtest/src/Common/src/System/Net/LazyAsyncResult.cs:line 26
2017-10-21 04:45:44,798: INFO: proc(54): run_and_log_output: Output:    at System.Net.LazyAsyncResult.Complete(IntPtr userToken) in /mnt/j/workspace/dotnet_corefx/master/linux-TGroup_netcoreapp+CGroup_Release+AGroup_x64+TestOuter_false_prtest/src/Common/src/System/Net/LazyAsyncResult.cs:line 404
2017-10-21 04:45:44,798: INFO: proc(54): run_and_log_output: Output:    at System.Net.Security.SslState.FinishHandshake(Exception e, AsyncProtocolRequest asyncRequest) in /mnt/j/workspace/dotnet_corefx/master/linux-TGroup_netcoreapp+CGroup_Release+AGroup_x64+TestOuter_false_prtest/src/System.Net.Security/src/System/Net/Security/SslState.cs:line 1454
2017-10-21 04:45:44,798: INFO: proc(54): run_and_log_output: Output:    at System.Net.Security.SslState.CheckCompletionBeforeNextReceive(ProtocolToken message, AsyncProtocolRequest asyncRequest) in /mnt/j/workspace/dotnet_corefx/master/linux-TGroup_netcoreapp+CGroup_Release+AGroup_x64+TestOuter_false_prtest/src/System.Net.Security/src/System/Net/Security/SslState.cs:line 839
2017-10-21 04:45:44,798: INFO: proc(54): run_and_log_output: Output:    at System.Net.Security.SslState.WriteCallback(IAsyncResult transportResult) in /mnt/j/workspace/dotnet_corefx/master/linux-TGroup_netcoreapp+CGroup_Release+AGroup_x64+TestOuter_false_prtest/src/System.Net.Security/src/System/Net/Security/SslState.cs:line 1107
2017-10-21 04:45:44,798: INFO: proc(54): run_and_log_output: Output:    at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state) in /root/coreclr/src/mscorlib/shared/System/Threading/ExecutionContext.cs:line 151
2017-10-21 04:45:44,799: INFO: proc(54): run_and_log_output: Output:    at System.Threading.Tasks.AwaitTaskContinuation.RunCallback(ContextCallback callback, Object state, Task& currentTask) in /root/coreclr/src/mscorlib/src/System/Threading/Tasks/TaskContinuation.cs:line 714
2017-10-21 04:45:44,799: INFO: proc(54): run_and_log_output: Output: --- End of stack trace from previous location where exception was thrown ---
2017-10-21 04:45:44,799: INFO: proc(54): run_and_log_output: Output:    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() in /root/coreclr/src/mscorlib/src/System/Runtime/ExceptionServices/ExceptionServicesCommon.cs:line 130
2017-10-21 04:45:44,799: INFO: proc(54): run_and_log_output: Output:    at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state) in /root/coreclr/src/mscorlib/shared/System/Threading/ExecutionContext.cs:line 151
2017-10-21 04:45:44,799: INFO: proc(54): run_and_log_output: Output:    at System.Threading.ThreadPoolWorkQueue.Dispatch() in /root/coreclr/src/mscorlib/src/System/Threading/ThreadPool.cs:line 588
```

**Comment by stephentoub** [comment]
@dotnet-bottest Outerloop Linux x64 Debug Build
@dotnet-bottest Outerloop Linux x64 Release Build

**Comment by stephentoub** [comment]
@jkotas, could this failure be related to your recent ThreadStatic changes?  The stack trace shows:
```
2017-10-21 04:45:44,798: INFO: proc(54): run_and_log_output: Output: Unhandled Exception: System.NullReferenceException: Object reference not set to an instance of an object.
2017-10-21 04:45:44,798: INFO: proc(54): run_and_log_output: Output:    at System.Net.LazyAsyncResult.get_CurrentThreadContext() in /mnt/j/workspace/dotnet_corefx/master/linux-TGroup_netcoreapp+CGroup_Release+AGroup_x64+TestOuter_false_prtest/src/Common/src/System/Net/LazyAsyncResult.cs:line 26
2017-10-21 04:45:44,798: INFO: proc(54): run_and_log_output: Output:    at System.Net.LazyAsyncResult.Complete(IntPtr userToken) in /mnt/j/workspace/dotnet_corefx/master/linux-TGroup_netcoreapp+CGroup_Release+AGroup_x64+TestOuter_false_prtest/src/Common/src/System/Net/LazyAsyncResult.cs:line 404
```
but get_CurrentThreadContext is:
https://github.com/dotnet/corefx/blob/ba635832fdd968ea0864fb0d7359e02ea5603725/src/Common/src/System/Net/LazyAsyncResult.cs#L22-L35
and I don't see how that could result in a null reference exception if everything was working properly.

**Comment by Drawaes** [comment]
I have the dumpling working thanks to @danmosemsft useful instructions. I can't really see anything obvious yet.

**Comment by jkotas** [comment]
This looks like the same problem as https://github.com/dotnet/corefx/issues/24722 . The managed thread statics get corrupted in System.Net.Security tests on Linux. @janvorli and @Priya91 has been looking into it.

It is interesting that we are only seeing the problem in System.Net.Security tests on Linux so far. I would expect managed thread static regression to show up in many different tests ...

**Comment by stephentoub** [comment]
Same crash just happened here:
https://ci3.dot.net/job/dotnet_corefx/job/master/job/linux-TGroup_netcoreapp+CGroup_Release+AGroup_x64+TestOuter_false_prtest/4596/
https://mc.dot.net/#/user/stephentoub/pr~2Fjenkins~2Fdotnet~2Fcorefx~2Fmaster~2F/test~2Ffunctional~2Fcli~2F/b428e1ace35a75c97d0c3d370ec345c299f1daa1/workItem/System.Net.Security.Tests/wilogs
```
2017-10-21 15:43:58,660: INFO: proc(54): run_and_log_output: Output: Unhandled Exception: System.NullReferenceException: Object reference not set to an instance of an object.
2017-10-21 15:43:58,660: INFO: proc(54): run_and_log_output: Output:    at System.Net.LazyAsyncResult.get_CurrentThreadContext() in /mnt/j/workspace/dotnet_corefx/master/linux-TGroup_netcoreapp+CGroup_Release+AGroup_x64+TestOuter_false_prtest/src/Common/src/System/Net/LazyAsyncResult.cs:line 26
2017-10-21 15:43:58,661: INFO: proc(54): run_and_log_output: Output:    at System.Net.LazyAsyncResult.Complete(IntPtr userToken) in /mnt/j/workspace/dotnet_corefx/master/linux-TGroup_netcoreapp+CGroup_Release+AGroup_x64+TestOuter_false_prtest/src/Common/src/System/Net/LazyAsyncResult.cs:line 404
2017-10-21 15:43:58,661: INFO: proc(54): run_and_log_output: Output:    at System.Net.Security.SslState.FinishHandshake(Exception e, AsyncProtocolRequest asyncRequest) in /mnt/j/workspace/dotnet_corefx/master/linux-TGroup_netcoreapp+CGroup_Release+AGroup_x64+TestOuter_false_prtest/src/System.Net.Security/src/System/Net/Security/SslState.cs:line 1454
2017-10-21 15:43:58,661: INFO: proc(54): run_and_log_output: Output:    at System.Net.Security.SslState.CheckCompletionBeforeNextReceive(ProtocolToken message, AsyncProtocolRequest asyncRequest) in /mnt/j/workspace/dotnet_corefx/master/linux-TGroup_netcoreapp+CGroup_Release+AGroup_x64+TestOuter_false_prtest/src/System.Net.Security/src/System/Net/Security/SslState.cs:line 839
2017-10-21 15:43:58,661: INFO: proc(54): run_and_log_output: Output:    at System.Net.Security.SslState.WriteCallback(IAsyncResult transportResult) in /mnt/j/workspace/dotnet_corefx/master/linux-TGroup_netcoreapp+CGroup_Release+AGroup_x64+TestOuter_false_prtest/src/System.Net.Security/src/System/Net/Security/SslState.cs:line 1107
2017-10-21 15:43:58,661: INFO: proc(54): run_and_log_output: Output:    at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state) in /root/coreclr/src/mscorlib/shared/System/Threading/ExecutionContext.cs:line 151
2017-10-21 15:43:58,661: INFO: proc(54): run_and_log_output: Output:    at System.Threading.Tasks.AwaitTaskContinuation.RunCallback(ContextCallback callback, Object state, Task& currentTask) in /root/coreclr/src/mscorlib/src/System/Threading/Tasks/TaskContinuation.cs:line 714
2017-10-21 15:43:58,661: INFO: proc(54): run_and_log_output: Output: --- End of stack trace from previous location where exception was thrown ---
2017-10-21 15:43:58,663: INFO: proc(54): run_and_log_output: Output:    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() in /root/coreclr/src/mscorlib/src/System/Runtime/ExceptionServices/ExceptionServicesCommon.cs:line 130
2017-10-21 15:43:58,663: INFO: proc(54): run_and_log_output: Output:    at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state) in /root/coreclr/src/mscorlib/shared/System/Threading/ExecutionContext.cs:line 151
2017-10-21 15:43:58,663: INFO: proc(54): run_and_log_output: Output:    at System.Threading.ThreadPoolWorkQueue.Dispatch() in /root/coreclr/src/mscorlib/src/System/Threading/ThreadPool.cs:line 588
2017-10-21 15:44:00,468: INFO: proc(54): run_and_log_output: Output: /home/helixbot/dotnetbuild/work/a8f6c3e1-882c-409f-bd91-7fff541d620e/Work/5b625004-0718-4f0c-9e30-8112fb60b460/Unzip/RunTests.sh: line 87:  1725 Aborted                 (core dumped) $RUNTIME_PATH/dotnet xunit.console.netcore.exe System.Net.Security.Tests.dll -xml testResults.xml -notrait Benchmark=true -notrait category=nonnetcoreapptests -notrait category=nonlinuxtests -notrait category=OuterLoop -notrait category=failing
2017-10-21 15:44:58,704: INFO: proc(54): run_and_log_output: Output: processing dump file /home/helixbot/dotnetbuild/work/a8f6c3e1-882c-409f-bd91-7fff541d620e/Work/5b625004-0718-4f0c-9e30-8112fb60b460/Unzip/core
2017-10-21 15:44:58,705: INFO: proc(54): run_and_log_output: Output: creating dumpling dump 8712226244430cb9a35b4aba164ca646a2b813ab
```

**Comment by Drawaes** [comment]
ref #24799

**Pull request closed without merging**